## EDA for DIDD project 
#### importing essential packages

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#### the following below gets rid of needing plt.show()

In [2]:
%matplotlib inline

#### importing the excel data with the specific sheets

In [3]:
eco_df = pd.read_excel ('..\Data\TEIS-NSS Project Data 10-2022.xlsx', sheet_name='ECO with Exit21-22')    

In [4]:
eco_df

DISTRICT  CHILD_ID  \
0                                                    ET  500335.0   
1                                                    ET  479453.0   
2                                                    ET  510663.0   
3                                                    ET  452482.0   
4                                                    ET  506507.0   
...                                                 ...       ...   
8631                                                 UC  514216.0   
8632                                                NaN       NaN   
8633                                                NaN       NaN   
8634                                                NaN       NaN   
8635  9/29/22 JM Corrected in TEIDS. Progress would ...       NaN   

     ECO_Entry_DATE  Ent SOCIAL_SCALE  Ent KNOWLEDGE_SCALE  \
0        2021-04-23               5.0                  3.0   
1        2020-03-23               7.0                  7.0   
2               NaT               NaN                  NaN   
3        2019-01-31               2.0                  3.0   
4               NaT               NaN                  NaN   
...             ...               ...                  ...   
8631            NaT               NaN                  NaN   
8632            NaT               NaN                  NaN   
8633            NaT               NaN                  NaN   
8634            NaT               NaN                  NaN   
8635            NaT               NaN                  NaN   

      Ent APPROPRIATE_ACTION_SCALE ECO_Exit_DATE  Exit SOCIAL_SCALE  \
0                              4.0    2021-10-23                6.0   
1                              7.0    2021-08-19                7.0   
2                              NaN           NaT                NaN   
3                              3.0    2021-11-22                5.0   
4                              NaN           NaT                NaN   
...                            ...           ...                ...   
8631                           NaN           NaT                NaN   
8632                           NaN           NaT                NaN   
8633                           NaN           NaT                NaN   
8634                           NaN           NaT                NaN   
8635                           NaN           NaT                NaN   

      Exit KNOWLEDGE_SCALE  Exit APPROPRIATE_ACTION_SCALE  ... OC2 Data Check  \
0                      3.0                            4.0  ...            1.0   
1                      7.0                            7.0  ...            1.0   
2                      NaN                            NaN  ...            0.0   
3                      3.0                            5.0  ...            1.0   
4                      NaN                            NaN  ...            0.0   
...                    ...                            ...  ...            ...   
8631                   NaN                            NaN  ...            0.0   
8632                   NaN                            NaN  ...            NaN   
8633                   NaN                            NaN  ...            NaN   
8634                   NaN                            NaN  ...            NaN   
8635                   NaN                            NaN  ...            NaN   

     OC3 - a OC3 - b OC3 - c OC3 - d OC3 - e OC3 Data Check  \
0        NaN     1.0     NaN     NaN     NaN            1.0   
1        NaN     NaN     NaN     NaN     1.0            1.0   
2        NaN     NaN     NaN     NaN     NaN            0.0   
3        NaN     NaN     1.0     NaN     NaN            1.0   
4        NaN     NaN     NaN     NaN     NaN            0.0   
...      ...     ...     ...     ...     ...            ...   
8631     NaN     NaN     NaN     NaN     NaN            0.0   
8632     NaN     NaN     NaN     NaN     NaN            NaN   
8633     NaN     NaN     NaN     NaN     NaN            NaN   
8634     NaN     

In [5]:
eco_df_check = eco_df['OC2 Data Check'].isin([1.0])
eco_df[eco_df_check]

DISTRICT  CHILD_ID ECO_Entry_DATE  Ent SOCIAL_SCALE  Ent KNOWLEDGE_SCALE  \
0          ET  500335.0     2021-04-23               5.0                  3.0   
1          ET  479453.0     2020-03-23               7.0                  7.0   
3          ET  452482.0     2019-01-31               2.0                  3.0   
5          ET  494216.0     2021-01-06               2.0                  4.0   
6          ET  449725.0     2020-08-06               4.0                  5.0   
...       ...       ...            ...               ...                  ...   
8624       UC  490337.0     2020-11-09               3.0                  1.0   
8626       UC  483729.0     2020-08-25               1.0                  3.0   
8627       UC  485978.0     2020-08-19               1.0                  3.0   
8628       UC  488802.0     2020-09-21               4.0                  4.0   
8630       UC  490688.0     2020-10-27               4.0                  5.0   

      Ent APPROPRIATE_ACTION_SCALE ECO_Exit_DATE  Exit SOCIAL_SCALE  \
0                              4.0    2021-10-23                6.0   
1                              7.0    2021-08-19                7.0   
3                              3.0    2021-11-22                5.0   
5                              6.0    2021-06-30                4.0   
6                              7.0    2021-04-21                7.0   
...                            ...           ...                ...   
8624                           1.0    2021-05-12                3.0   
8626                           2.0    2021-08-12                6.0   
8627                           5.0    2021-02-11                5.0   
8628                           3.0    2021-09-09                6.0   
8630                           7.0    2021-10-13                7.0   

      Exit KNOWLEDGE_SCALE  Exit APPROPRIATE_ACTION_SCALE  ... OC2 Data Check  \
0                      3.0                            4.0  ...            1.0   
1                      7.0                            7.0  ...            1.0   
3                      3.0                            5.0  ...            1.0   
5                      5.0                            7.0  ...            1.0   
6                      7.0                            7.0  ...            1.0   
...                    ...                            ...  ...            ...   
8624                   3.0                            1.0  ...            1.0   
8626                   6.0                            6.0  ...            1.0   
8627                   7.0                            7.0  ...            1.0   
8628                   7.0                            7.0  ...            1.0   
8630                   7.0                            7.0  ...            1.0   

     OC3 - a OC3 - b OC3 - c OC3 - d OC3 - e OC3 Data Check  \
0        NaN     1.0     NaN     NaN     NaN            1.0   
1        NaN     NaN     NaN     NaN     1.0            1.0   
3        NaN     NaN     1.0     NaN     NaN            1.0   
5        NaN     NaN     NaN     NaN     1.0            1.0   
6        NaN     NaN     NaN     NaN     1.0            1.0   
...      ...     ...     ...     ...     ...            ...   
8624     NaN     1.0     NaN     NaN     NaN            1.0   
8626     NaN     NaN     NaN     1.0     NaN            1.0   
8627     NaN     NaN     NaN     1.0     NaN            1.0   
8628     NaN     NaN     NaN     1.0     NaN            1.0   
8630     NaN     NaN     NaN     NaN     1.0            1.0   

      Days btw Initial and Exit  Days btw I-IFSP to Exit ECO  \
0                         206.0                        183.0   
1                         704.0                        514.0   
3                        1053.0                       1026.0   
5                         288.0                        175.0   
6                         444.0                        348.0   
...                         ...                          ...   
8

In [6]:
elig_df = pd.read_excel ('..\Data\TEIS-NSS Project Data 10-2022.xlsx', sheet_name = 'Elig Timeline Rpt 2018-2022') 

In [7]:
elig_df.head()

District  Child ID Child Status  Child Phase Init. IFSP Due Date  \
0       ET    453926     Inactive         IFSP          2019-03-04   
1       ET    431729     Inactive         IFSP          2018-10-26   
2       ET    462474     Inactive         IFSP          2019-06-28   
3       ET    446841     Inactive  Eligibility          2018-11-23   
4       ET    459629     Inactive         IFSP          2019-05-16   

  IFSP Due Date IFSP Late Reason Active Ref. Date Parent Consent/Intake Date  \
0    2019-03-27           System       2019-01-18                 2019-02-04   
1    2018-10-26              NaN       2018-09-11                 2018-09-25   
2    2019-06-14              NaN       2019-05-14                 2019-05-30   
3           NaT              NaN       2018-10-09                 2018-10-25   
4    2019-04-24              NaN       2019-04-01                 2019-04-10   

  Date Dev. Evaluator Rec'd. Ref.  ... # Days Parent Consent to Dev. Eval.  \
0             2019-01-22 00:00:00  ...                                17.0   
1             2018-09-12 00:00:00  ...                                 0.0   
2             2019-05-15 00:00:00  ...                                 0.0   
3             2018-10-11 00:00:00  ...                                 0.0   
4             2019-04-02 00:00:00  ...                                 0.0   

  # Days Dev. Evaluator Assigned to Dev. Eval.  \
0                                         30.0   
1                                         13.0   
2                                         15.0   
3                                         14.0   
4                                          8.0   

  1st Date Requested to Receipt of Med. Records  \
0                                           NaN   
1                                           NaN   
2                                           0.0   
3                                           NaN   
4                                           0.0   

  # Days Dev. Eval. to Elig. Det. # Days Ref. to Elig. Det.  \
0                             4.0                      38.0   
1                             6.0                      20.0   
2                             1.0                      17.0   
3                             1.0                      17.0   
4                             2.0                      11.0   

  Init. Elig. Decision                           Init. Elig. Category  \
0             Eligible                       Developmental Evaluation   
1             Eligible  Diagnosed Condition, Developmental Evaluation   
2             Eligible                       Developmental Evaluation   
3             Eligible                            Diagnosed Condition   
4             Eligible                       Developmental Evaluation   

  Init. Elig. Det. Date  Exit Date                              Exit Reason  
0            2019-02-25 2020-01-11                    618 - Part B eligible  
1            2018-10-01 2020-12-11  618 - Part B eligibility not determined  
2            2019-05-31 2020-11-12  618 - Part B eligibility not determined  
3            2018-10-26 2020-08-12                           Parent decline  
4            2019-04-12 2019-09-26                    618 - Parent withdraw  

[5 rows x 38 columns]

In [8]:
eco_df.rename (columns = {'CHILD_ID':'Child ID'}, inplace=True )
eco_df

DISTRICT  Child ID  \
0                                                    ET  500335.0   
1                                                    ET  479453.0   
2                                                    ET  510663.0   
3                                                    ET  452482.0   
4                                                    ET  506507.0   
...                                                 ...       ...   
8631                                                 UC  514216.0   
8632                                                NaN       NaN   
8633                                                NaN       NaN   
8634                                                NaN       NaN   
8635  9/29/22 JM Corrected in TEIDS. Progress would ...       NaN   

     ECO_Entry_DATE  Ent SOCIAL_SCALE  Ent KNOWLEDGE_SCALE  \
0        2021-04-23               5.0                  3.0   
1        2020-03-23               7.0                  7.0   
2               NaT               NaN                  NaN   
3        2019-01-31               2.0                  3.0   
4               NaT               NaN                  NaN   
...             ...               ...                  ...   
8631            NaT               NaN                  NaN   
8632            NaT               NaN                  NaN   
8633            NaT               NaN                  NaN   
8634            NaT               NaN                  NaN   
8635            NaT               NaN                  NaN   

      Ent APPROPRIATE_ACTION_SCALE ECO_Exit_DATE  Exit SOCIAL_SCALE  \
0                              4.0    2021-10-23                6.0   
1                              7.0    2021-08-19                7.0   
2                              NaN           NaT                NaN   
3                              3.0    2021-11-22                5.0   
4                              NaN           NaT                NaN   
...                            ...           ...                ...   
8631                           NaN           NaT                NaN   
8632                           NaN           NaT                NaN   
8633                           NaN           NaT                NaN   
8634                           NaN           NaT                NaN   
8635                           NaN           NaT                NaN   

      Exit KNOWLEDGE_SCALE  Exit APPROPRIATE_ACTION_SCALE  ... OC2 Data Check  \
0                      3.0                            4.0  ...            1.0   
1                      7.0                            7.0  ...            1.0   
2                      NaN                            NaN  ...            0.0   
3                      3.0                            5.0  ...            1.0   
4                      NaN                            NaN  ...            0.0   
...                    ...                            ...  ...            ...   
8631                   NaN                            NaN  ...            0.0   
8632                   NaN                            NaN  ...            NaN   
8633                   NaN                            NaN  ...            NaN   
8634                   NaN                            NaN  ...            NaN   
8635                   NaN                            NaN  ...            NaN   

     OC3 - a OC3 - b OC3 - c OC3 - d OC3 - e OC3 Data Check  \
0        NaN     1.0     NaN     NaN     NaN            1.0   
1        NaN     NaN     NaN     NaN     1.0            1.0   
2        NaN     NaN     NaN     NaN     NaN            0.0   
3        NaN     NaN     1.0     NaN     NaN            1.0   
4        NaN     NaN     NaN     NaN     NaN            0.0   
...      ...     ...     ...     ...     ...            ...   
8631     NaN     NaN     NaN     NaN     NaN            0.0   
8632     NaN     NaN     NaN     NaN     NaN            NaN   
8633     NaN     NaN     NaN     NaN     NaN            NaN   
8634     NaN     

## merging the data frames

In [9]:
eco_elig_df = pd.merge(eco_df, elig_df, how="outer", on= "Child ID")
eco_elig_df

DISTRICT  Child ID ECO_Entry_DATE  Ent SOCIAL_SCALE  \
0           ET  500335.0     2021-04-23               5.0   
1           ET  479453.0     2020-03-23               7.0   
2           ET  510663.0            NaT               NaN   
3           ET  452482.0     2019-01-31               2.0   
4           ET  506507.0            NaT               NaN   
...        ...       ...            ...               ...   
67609      NaN  514221.0            NaT               NaN   
67610      NaN  525915.0            NaT               NaN   
67611      NaN  516841.0            NaT               NaN   
67612      NaN  526338.0            NaT               NaN   
67613      NaN  511641.0            NaT               NaN   

       Ent KNOWLEDGE_SCALE  Ent APPROPRIATE_ACTION_SCALE ECO_Exit_DATE  \
0                      3.0                           4.0    2021-10-23   
1                      7.0                           7.0    2021-08-19   
2                      NaN                           NaN           NaT   
3                      3.0                           3.0    2021-11-22   
4                      NaN                           NaN           NaT   
...                    ...                           ...           ...   
67609                  NaN                           NaN           NaT   
67610                  NaN                           NaN           NaT   
67611                  NaN                           NaN           NaT   
67612                  NaN                           NaN           NaT   
67613                  NaN                           NaN           NaT   

       Exit SOCIAL_SCALE  Exit KNOWLEDGE_SCALE  Exit APPROPRIATE_ACTION_SCALE  \
0                    6.0                   3.0                            4.0   
1                    7.0                   7.0                            7.0   
2                    NaN                   NaN                            NaN   
3                    5.0                   3.0                            5.0   
4                    NaN                   NaN                            NaN   
...                  ...                   ...                            ...   
67609                NaN                   NaN                            NaN   
67610                NaN                   NaN                            NaN   
67611                NaN                   NaN                            NaN   
67612                NaN                   NaN                            NaN   
67613                NaN                   NaN                            NaN   

       ... # Days Parent Consent to Dev. Eval.  \
0      ...                                 7.0   
1      ...                                -1.0   
2      ...                                 0.0   
3      ...                                 0.0   
4      ...                                10.0   
...    ...                                 ...   
67609  ...                                 8.0   
67610  ...                                 1.0   
67611  ...                                18.0   
67612  ...                                 1.0   
67613  ...                                 0.0   

      # Days Dev. Evaluator Assigned to Dev. Eval.  \
0                                              7.0   
1                                              7.0   
2                                              7.0   
3                                              NaN   
4                                             31.0   
...                                            ...   
67609                                         11.0   
67610                                          6.0   
67611                                         22.0   
67612                                          3.0   
67613                                          7.0   

      1st Date Requested to Receipt of Med. Records  \
0                                               NaN   
1                                               0.0

In [10]:
eco_elig_check = eco_elig_df['OC2 Data Check'].isin([1.0])
EEC2 = pd.DataFrame(eco_elig_df[eco_elig_check])

EEC2

DISTRICT  Child ID ECO_Entry_DATE  Ent SOCIAL_SCALE  Ent KNOWLEDGE_SCALE  \
0          ET  500335.0     2021-04-23               5.0                  3.0   
1          ET  479453.0     2020-03-23               7.0                  7.0   
3          ET  452482.0     2019-01-31               2.0                  3.0   
5          ET  494216.0     2021-01-06               2.0                  4.0   
6          ET  449725.0     2020-08-06               4.0                  5.0   
...       ...       ...            ...               ...                  ...   
8624       UC  490337.0     2020-11-09               3.0                  1.0   
8626       UC  483729.0     2020-08-25               1.0                  3.0   
8627       UC  485978.0     2020-08-19               1.0                  3.0   
8628       UC  488802.0     2020-09-21               4.0                  4.0   
8630       UC  490688.0     2020-10-27               4.0                  5.0   

      Ent APPROPRIATE_ACTION_SCALE ECO_Exit_DATE  Exit SOCIAL_SCALE  \
0                              4.0    2021-10-23                6.0   
1                              7.0    2021-08-19                7.0   
3                              3.0    2021-11-22                5.0   
5                              6.0    2021-06-30                4.0   
6                              7.0    2021-04-21                7.0   
...                            ...           ...                ...   
8624                           1.0    2021-05-12                3.0   
8626                           2.0    2021-08-12                6.0   
8627                           5.0    2021-02-11                5.0   
8628                           3.0    2021-09-09                6.0   
8630                           7.0    2021-10-13                7.0   

      Exit KNOWLEDGE_SCALE  Exit APPROPRIATE_ACTION_SCALE  ...  \
0                      3.0                            4.0  ...   
1                      7.0                            7.0  ...   
3                      3.0                            5.0  ...   
5                      5.0                            7.0  ...   
6                      7.0                            7.0  ...   
...                    ...                            ...  ...   
8624                   3.0                            1.0  ...   
8626                   6.0                            6.0  ...   
8627                   7.0                            7.0  ...   
8628                   7.0                            7.0  ...   
8630                   7.0                            7.0  ...   

     # Days Parent Consent to Dev. Eval.  \
0                                    7.0   
1                                   -1.0   
3                                    0.0   
5                                    1.0   
6                                   38.0   
...                                  ...   
8624                                 0.0   
8626                                 1.0   
8627                                 0.0   
8628                                 0.0   
8630                                 1.0   

     # Days Dev. Evaluator Assigned to Dev. Eval.  \
0                                             7.0   
1                                             7.0   
3                                             NaN   
5                                             8.0   
6                                            51.0   
...                                           ...   
8624                                         16.0   
8626                                          1.0   
8627                                          8.0   
8628                                          6.0   
8630                                          2.0   

     1st Date Requested to Receipt of Med. Records  \
0                                              NaN   
1                                              0.0   
3                                              NaN   
5   

In [11]:
EEC2.count(axis = 0)

DISTRICT                 5595
Child ID                 5595
ECO_Entry_DATE           5595
Ent SOCIAL_SCALE         5595
Ent KNOWLEDGE_SCALE      5595
                         ... 
Init. Elig. Decision     5595
Init. Elig. Category     5594
Init. Elig. Det. Date    5595
Exit Date_y              5572
Exit Reason_y            5572
Length: 97, dtype: int64

In [12]:
eco_elig_check = eco_elig_df['OC1 Data Check'].isin([1.0])
EEC1 = pd.DataFrame(eco_elig_df[eco_elig_check])

EEC1.shape

(5595, 97)

In [13]:
eco_elig_check = eco_elig_df['OC3 Data Check'].isin([1.0])
EEC3 = pd.DataFrame(eco_elig_df[eco_elig_check])

EEC3.shape

(5595, 97)

In [14]:
# Outcome 1 counts 
EEC1['Init. Elig. Category'].value_counts()

Developmental Evaluation                         4074
Diagnosed Condition, Developmental Evaluation     854
Diagnosed Condition                               666
Name: Init. Elig. Category, dtype: int64

In [15]:
# Outcome 2 counts
EEC2['Init. Elig. Category'].value_counts()

Developmental Evaluation                         4074
Diagnosed Condition, Developmental Evaluation     854
Diagnosed Condition                               666
Name: Init. Elig. Category, dtype: int64

In [16]:
# Outcome 3 counts
EEC3['Init. Elig. Category'].value_counts() 

Developmental Evaluation                         4074
Diagnosed Condition, Developmental Evaluation     854
Diagnosed Condition                               666
Name: Init. Elig. Category, dtype: int64

In [17]:
#getting the percentages 
round((EEC3['Init. Elig. Category'].value_counts()/EEC3['Init. Elig. Category'].value_counts().sum())*100,2)

Developmental Evaluation                         72.83
Diagnosed Condition, Developmental Evaluation    15.27
Diagnosed Condition                              11.91
Name: Init. Elig. Category, dtype: float64

#### grabbing info on the three categories of eligibility

In [18]:
NA_EEC2 = EEC2['Init. Elig. Category'].isna()
EEC2[NA_EEC2]
# child ID 468726.0 (id should be an INT, not a float)

DISTRICT  Child ID ECO_Entry_DATE  Ent SOCIAL_SCALE  Ent KNOWLEDGE_SCALE  \
6646       SC  468726.0     2021-04-21               7.0                  7.0   

      Ent APPROPRIATE_ACTION_SCALE ECO_Exit_DATE  Exit SOCIAL_SCALE  \
6646                           7.0    2021-08-21                7.0   

      Exit KNOWLEDGE_SCALE  Exit APPROPRIATE_ACTION_SCALE  ...  \
6646                   7.0                            7.0  ...   

     # Days Parent Consent to Dev. Eval.  \
6646                                 0.0   

     # Days Dev. Evaluator Assigned to Dev. Eval.  \
6646                                        545.0   

     1st Date Requested to Receipt of Med. Records  \
6646                                           NaN   

     # Days Dev. Eval. to Elig. Det. # Days Ref. to Elig. Det.  \
6646                            24.0                      31.0   

     Init. Elig. Decision Init. Elig. Category  Init. Elig. Det. Date  \
6646           Ineligible                  NaN             2021-03-19   

      Exit Date_y          Exit Reason_y  
6646   2021-11-22  618 - Part B eligible  

[1 rows x 97 columns]

<bold> Developmental Evaluation

In [19]:
elig_cat_DE = EEC2[EEC2['Init. Elig. Category'] == 'Developmental Evaluation']

In [20]:
elig_cat_DE.shape

(4074, 97)

<bold> Diagnosed Condition

In [21]:
elig_cat_DC = EEC2[EEC2['Init. Elig. Category'] == 'Diagnosed Condition']

In [22]:
elig_cat_DC.shape

(666, 97)

<bold> BOTH: Developmental Evaluation & Diagnosed Condition

In [23]:
elig_cat_both = EEC2[EEC2['Init. Elig. Category'] == 'Diagnosed Condition, Developmental Evaluation']

In [24]:
elig_cat_both.shape

(854, 97)

In [25]:
EEC1['Exit Reason_y'] = EEC1['Exit Reason_y'].str.strip('618 - ')
EEC1.head()

DISTRICT  Child ID ECO_Entry_DATE  Ent SOCIAL_SCALE  Ent KNOWLEDGE_SCALE  \
0       ET  500335.0     2021-04-23               5.0                  3.0   
1       ET  479453.0     2020-03-23               7.0                  7.0   
3       ET  452482.0     2019-01-31               2.0                  3.0   
5       ET  494216.0     2021-01-06               2.0                  4.0   
6       ET  449725.0     2020-08-06               4.0                  5.0   

   Ent APPROPRIATE_ACTION_SCALE ECO_Exit_DATE  Exit SOCIAL_SCALE  \
0                           4.0    2021-10-23                6.0   
1                           7.0    2021-08-19                7.0   
3                           3.0    2021-11-22                5.0   
5                           6.0    2021-06-30                4.0   
6                           7.0    2021-04-21                7.0   

   Exit KNOWLEDGE_SCALE  Exit APPROPRIATE_ACTION_SCALE  ...  \
0                   3.0                            4.0  ...   
1                   7.0                            7.0  ...   
3                   3.0                            5.0  ...   
5                   5.0                            7.0  ...   
6                   7.0                            7.0  ...   

  # Days Parent Consent to Dev. Eval.  \
0                                 7.0   
1                                -1.0   
3                                 0.0   
5                                 1.0   
6                                38.0   

  # Days Dev. Evaluator Assigned to Dev. Eval.  \
0                                          7.0   
1                                          7.0   
3                                          NaN   
5                                          8.0   
6                                         51.0   

  1st Date Requested to Receipt of Med. Records  \
0                                           NaN   
1                                           0.0   
3                                           NaN   
5                                           0.0   
6                                           NaN   

  # Days Dev. Eval. to Elig. Det. # Days Ref. to Elig. Det.  \
0                             9.0                      20.0   
1                             8.0                      15.0   
3                             1.0                      14.0   
5                             0.0                       9.0   
6                             2.0                      55.0   

  Init. Elig. Decision      Init. Elig. Category  Init. Elig. Det. Date  \
0             Eligible  Developmental Evaluation             2021-04-14   
1             Eligible  Developmental Evaluation             2020-03-20   
3             Eligible  Developmental Evaluation             2019-01-16   
5             Eligible  Developmental Evaluation             2020-12-17   
6             Eligible  Developmental Evaluation             2020-05-06   

   Exit Date_y                                    Exit Reason_y  
0   2021-11-15                                  Part B eligible  
1   2022-02-25  Completion of IFSP/ no longer requires services  
3   2021-12-19                                  Part B eligible  
5   2021-10-21                                  Parent withdraw  
6   2021-07-26                                  Parent withdraw  

[5 rows x 97 columns]

#### So...out of the 5595 kids for OC2 check <br>
#### 4074 were categorized as Development Evaluation <br>
#### 666 were categorized as Diagnosed Condition <br>
#### 854 were categorized as both <br>
#### this leaves 1 kid unaccounted for 

# EXIT REASONS

#### Do exit reasons vary by eligibility category? 

In [26]:
EEC1['Exit Reason_y'].value_counts().sum()

5572

In [27]:
EEC1['Exit Reason_y'].value_counts()

Part B eligible                                                   1741
Part B eligibility not determined                                 1577
Parent withdraw                                                   1134
Not eligible for Part B – Exit with referral to other programs     308
Attempts to contact unsuccessful                                   299
Not eligible for Part B – Exit with no referrals                   298
Moved out of state                                                 148
Completion of IFSP/ no longer requires services                     57
Deceased                                                            10
Name: Exit Reason_y, dtype: int64

### Developmental Evaluation

In [28]:
DE = EEC1[EEC1['Init. Elig. Category'] == 'Developmental Evaluation']
DE['Exit Reason_y'].value_counts()
#Part B eligible, #1

Part B eligible                                                   1318
Part B eligibility not determined                                 1182
Parent withdraw                                                    780
Not eligible for Part B – Exit with no referrals                   231
Not eligible for Part B – Exit with referral to other programs     227
Attempts to contact unsuccessful                                   185
Moved out of state                                                  93
Completion of IFSP/ no longer requires services                     38
Deceased                                                             4
Name: Exit Reason_y, dtype: int64

In [29]:
DE['Exit Reason_y'].value_counts().sum()

4058

### Diagnosed Condition

In [30]:
DC = EEC1[EEC1['Init. Elig. Category'] == 'Diagnosed Condition']
DC['Exit Reason_y'].value_counts()
#Part B eligibility not determined, #1

Part B eligibility not determined                                 183
Part B eligible                                                   172
Parent withdraw                                                   156
Attempts to contact unsuccessful                                   52
Not eligible for Part B – Exit with referral to other programs     36
Not eligible for Part B – Exit with no referrals                   33
Moved out of state                                                 19
Completion of IFSP/ no longer requires services                    12
Deceased                                                            1
Name: Exit Reason_y, dtype: int64

In [31]:
DC['Exit Reason_y'].value_counts().sum()

664

### BOTH: Developmental Evaluation & Diagnosed Condition

In [32]:
DC_DE = EEC1[EEC1['Init. Elig. Category'] == 'Diagnosed Condition, Developmental Evaluation']
DC_DE['Exit Reason_y'].value_counts()
# Part B eligible, #1
#Parent withdraw was #3 for all 

Part B eligible                                                   250
Part B eligibility not determined                                 212
Parent withdraw                                                   198
Attempts to contact unsuccessful                                   62
Not eligible for Part B – Exit with referral to other programs     45
Moved out of state                                                 36
Not eligible for Part B – Exit with no referrals                   34
Completion of IFSP/ no longer requires services                     7
Deceased                                                            5
Name: Exit Reason_y, dtype: int64

In [33]:
DC_DE['Exit Reason_y'].value_counts().sum()

849

#### DE - 4058 <br>
#### DC - 664 <br>
#### DC_DE - 849 <br>

In [34]:
num = [4058, 664, 849]
sum(num)
#there are 23 nulls kid so if you count those 5594 

5571

To answer the first part, I think the exit reasons do not vary extremely from one eligibility category to the next. There are some variations but slight.

#### Do more children in one eligibility category age out compared to leaving for other reasons?

They all typically aged out....looking at Part B eligible (and eligibility not determined)

### ------------------------------------------------------------------

#### Box Plots

Boxplots also give you a lot of information about a continuous variable in your dataset. The middle line through the box indicates the median value. The top and botton sides of the box are the 75th and 25th percentiles (3rd and 4th quartiles). The bars above and below the box (called whiskers) indicated the bounds for the upper and lower extremes of the distribution. Any points (values) beyond the whiskers are considered outliers